astropy/tests/helper.py

```diff
@@ -604,3 +604,42 @@ def check_pickling_recovery(original, protocol):
     class_history = [original.__class__]
     generic_recursive_equality_test(original, unpickled,
                                     class_history)
+
+
+def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=0, err_msg='', verbose=True):
+    """
+    Raise an assertion if two objects are not equal up to desired tolerance.
+
+    This is a :class:`~astropy.units.Quantity`-aware version of
+    :func:`numpy.testing.assert_allclose`.
+    """
+
+    import numpy as np
+    from .. import units as u
+
+    actual = u.Quantity(actual, subok=True, copy=False)
+
+    desired = u.Quantity(desired, subok=True, copy=False)
+    try:
+        desired = desired.to(actual.unit)
+    except u.UnitsError:
+        raise u.UnitsError("Units for 'desired' ({0}) and 'actual' ({1}) are not convertible".format(desired.unit, actual.unit))
+
+    if atol == 0:
+        atol = u.Quantity(0)
```

Comment:
Tiny really, but somehow this feels slightly confusing. How about writing
```
# we take 0 to be convertible to any unit
atol = u.Quantity(0, actual.unit)
```
You could also consider making the default `atol=None` and then test for that, i.e.,
```
if atol is None:
    # by default, we assume an absolute tolerance of 0
    atol = u.Quantity(0, actual.unit)
else:
    ...
```